### PR TITLE
[#3092] fix(relational-entity-store): Return false when deleting throw a `NoSuchEntityException`

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

When Relational Entity Store deleteing an entity and throwing a NoSuchEntityException, return false finally. It's consistent with KV Store.

### Why are the changes needed?

Fix: #3092 

### How was this patch tested?

Add some duplicate deleted uts.

### CODE REVIEW

**Comment by xloya** [comment]
cc @jerryshao @yuqi1129

**Comment by xloya** [comment]
Another thing to confirm is that in the `Relational Entity Store`, there are two situations where `NoSuchEntityException` will be returned:
1. If it is found that the parent entity does not exist, throw `NoSuchEntityException`;
2. If it is found that the entity that needs to be operated does not exist, throw `NoSuchEntityException`;
I'm not sure returning false for both cases is consistent with the `KV Store`.

**Comment by jerryshao** [comment]
@yuqi1129 can you please confirm this?

**Comment by yuqi1129** [comment]
> @yuqi1129 can you please confirm this?

Generally, I'm okay with the changes. The logic is consistent with the KV store. @jerryshao WDYT

**Comment by jerryshao** [comment]
I'm fine if the semantics are consistent. @yuqi1129 @qqqttt123 please review the detailed implementation to see if it is OK or not.

**Comment by qqqttt123** [comment]
Maybe this is a fix instead of an improvement.

File: core/src/test/java/com/datastrato/gravitino/storage/TestEntityStorage.java [hunk 1]
```diff
@@ -1567,6 +1598,8 @@ private void validateDeleteUser(EntityStore store, UserEntity user1) throws IOEx
     Assertions.assertTrue(store.exists(user1.nameIdentifier(), Entity.EntityType.USER));
     Assertions.assertTrue(store.delete(user1.nameIdentifier(), Entity.EntityType.USER));
     Assertions.assertFalse(store.exists(user1.nameIdentifier(), Entity.EntityType.USER));
+    // delete again should return false
+    Assertions.assertFalse(store.delete(user1.nameIdentifier(), Entity.EntityType.USER));
```
Comment (qqqttt123) [inline core/src/test/java/com/datastrato/gravitino/storage/TestEntityStorage.java:1602]: Do we need similar test cases for groups and roles?

Comment (xloya) [reply core/src/test/java/com/datastrato/gravitino/storage/TestEntityStorage.java:1602]: Yes, I think it is necessary, please let me update the pr.

Comment (xloya) [reply core/src/test/java/com/datastrato/gravitino/storage/TestEntityStorage.java:1602]: Done

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**